velox/row/tests/UnsafeRowSerdeTest.cpp

```diff
@@ -0,0 +1,96 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+
+#include "velox/row/UnsafeRowDeserializers.h"
+
+#include "velox/row/UnsafeRowFast.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::velox;
+
+namespace facebook::velox::row {
+class UnsafeRowSerdeTests : public testing::Test, public test::VectorTestBase {
+ public:
+  UnsafeRowSerdeTests() {}
+
+  RowVectorPtr createInputRowWithUnknownType(
+      int32_t batchSize,
+      std::function<bool(vector_size_t /*row*/)> isNullAt = nullptr) {
+    VELOX_CHECK(batchSize <= kMaxBuffers);
+    auto flatVector =
+        makeFlatVector<UnknownValue>(
+            batchSize,
+            [](vector_size_t i) { return UnknownValue(); },
+            [](vector_size_t i) { return true; });
+    auto arrayVector = makeArrayVector<UnknownValue>(
+        batchSize,
+        [](vector_size_t row) { return row % 5; },
+        [](vector_size_t row, vector_size_t index) {
+          return UnknownValue();
+        },
+        [](vector_size_t i) { return true; });
+
+    auto mapVector = makeMapVector<UnknownValue, UnknownValue>(
+        batchSize,
+        [&](vector_size_t /*row*/) { return 1; },
+        [&](vector_size_t row) { return UnknownValue(); },
+        [&](vector_size_t row) { return UnknownValue(); },
+        [](vector_size_t i) { return true; },
+        [](vector_size_t i) { return true; });
+    auto rowVector = makeRowVector({flatVector});
+    return makeRowVector(
+        {arrayVector, mapVector, flatVector, rowVector}, isNullAt);
+  }
+
+  void testVectorSerde(const RowVectorPtr& inputVector) {
+    std::vector<std::optional<std::string_view>> serializedVector;
+    serializedVector.reserve(inputVector->size());
+    UnsafeRowFast fast(inputVector);
+
+    for (size_t i = 0; i < inputVector->size(); ++i) {
+      // Serialize rowVector into bytes.
+      auto rowSize = fast.serialize(i, buffers_[i]);
+
+      EXPECT_EQ(rowSize, fast.rowSize(i));
+
+      if (rowSize) {
+        serializedVector.push_back(
+            std::string_view(buffers_[i], rowSize));
+      } else {
+        serializedVector.push_back(std::nullopt);
+      }
+    }
+    VectorPtr outputVector = UnsafeRowDeserializer::deserialize(
+        serializedVector, inputVector->type(), pool());
+    test::assertEqualVectors(inputVector, outputVector);
+  }
+
+ private:
+  constexpr static int kMaxBuffers = 10;
+  std::array<char[1024], kMaxBuffers> buffers_{};
```

Comment:
Let's not hard-code size of buffer and number of buffer, but allocate on-demand.